crates/provider/src/lib.rs

```diff
@@ -15,11 +15,19 @@
 #![deny(unused_must_use, rust_2018_idioms)]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
 
-use alloy_transport_http::Http;
-use reqwest::Client as ReqwestClient;
+#[cfg(feature = "http")]
+mod http {
+    use alloy_transport_http::Http;
+    use reqwest::Client as ReqwestClient;
```

Comment:
this alias is kinda pointless, let's just use reqwest::clinet in the pub type HttpProvider